src/linter/conf.go

```diff
@@ -68,7 +68,7 @@ func NewConfig() *Config {
 	addBuiltinCheckers(reg)
 	return &Config{
 		SrcInput:       inputs.NewDefaultSourceInput(),
-		Rules:          &rules.Set{},
+		Rules:          rules.NewSet(),
```

Comment:
There is a problem here.
cloneRulesForFile from linter/utils.go will now copy rules sets even if rules are not used.
We need to change `ruleSet != nil` check with something else as they're never nil now.